### DISCUSSION TITLE
Rendering 15K records

### DISCUSSION THREAD

**Discussion by Shyam3327Agarwal** [l-lin/angular-datatables#1661] [Q&A] [answered]
Hi,

Actually there is a scenario where I need to render about 15K records and I don't have the option to use server side rendering.

So, basically I am using the ngFor and then via trigger I am able to render the table but the problem I am facing is datatable take a lot of time get render and the page becomes laggy as well.

Apart from the above issue the other thing I am facing is if I use a custom search field for my datatable that takes some extra time to search as compared to the default search I get.  

Thanks for the help in advance.

**Comment by shanmukhateja** [comment, marked as answer]
Hello,

We cannot expect the table or browser to handle 15K records on the DOM **at all times**. This is why we have server-side processing.

I highly recommend implementing [server-side processing](https://datatables.net/manual/server-side#Sent-parameters) or [infinite scroll](https://datatables.net/extensions/scroller/) as there's no other way around it.

Hope this helps :)
